servicetalk-concurrent-api-test/src/test/java/io/servicetalk/concurrent/api/test/CompletableStepVerifierTest.java

```diff
@@ -0,0 +1,285 @@
+                        }
+                    })
+                    .expectComplete()
+                    .verify(ofNanos(10)));
+        } finally {
+            latch.countDown();
+        }
+    }
+
+    @Test
+    public void onComplete() {
+        assertNotNull(create(completed())
+                .expectComplete()
+                .verify());
+    }
+
+    @Test
+    public void onCompleteDuplicateVerify() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(2);
+        StepVerifier verifier = create(completed())
+                .expectCancellable(cancellable -> {
+                    assertNotNull(cancellable);
+                    latch.countDown();
+                })
+                .expectComplete();
+        verifier.verify();
+        verifier.verify();
+        assertTrue(latch.await(10, SECONDS));
+    }
+
+    @Test
+    public void onCompleteLargeTimeout() {
+        assertNotNull(create(completed())
+                .expectComplete()
+                .verify(ofDays(1)));
+    }
+
+    @Test(expected = AssertionError.class)
+    public void onCompleteTimeout() {
+        verifyException(() -> create(never())
+                .expectComplete()
+                .verify(ofNanos(10)));
+    }
+
+    @Test
+    public void onErrorClass() {
+        create(failed(DELIBERATE_EXCEPTION))
+                .expectError(DeliberateException.class)
+                .verify();
+    }
+
+    @Test
+    public void onErrorPredicate() {
+        create(failed(DELIBERATE_EXCEPTION))
+                .expectError(error -> error instanceof DeliberateException)
+                .verify();
+    }
+
+    @Test
+    public void onErrorConsumer() {
+        create(failed(DELIBERATE_EXCEPTION))
+                .expectError(error -> {
+                    assertThat(error, is(DELIBERATE_EXCEPTION));
+                })
+                .verify();
+    }
+
+    @Test(expected = AssertionError.class)
+    public void expectOnErrorWhenOnComplete() {
+        verifyException(() -> create(completed())
+                .expectError(DeliberateException.class)
+                .verify());
+    }
+
+    @Test(expected = AssertionError.class)
+    public void expectOnSuccessWhenOnError() {
+        verifyException(() -> create(failed(DELIBERATE_EXCEPTION))
+                    .expectComplete()
+                    .verify());
+    }
+
+    @Test
+    public void noSignalsSubscriptionCancelSucceeds() {
+        // expectNoSignals and subscription event are dequeued/processed sequentially on the Subscriber thread
+        // and the scenario isn't instructed to expect the subscription so we pass the test.
+        create(never())
+                .expectNoSignals(ofDays(1))
+                .thenCancel()
+                .verify();
+    }
+
+    @Test(expected = AssertionError.class)
+    public void noSignalsCompleteFail() {
+        verifyException(() -> create(completed())
+                .expectCancellable(c -> { })
```

Comment:
This isn't necessary as the test utilities will fail if there is no `Subscription` before other signals are delivered because this is never valid per RS spec.

Reply:
Does that mean that we can remove all `.expectCancellable(c -> { })` and `.expectCancellable(Assert::assertNotNull)` from all existing tests?

Reply:
After thinking about it the `expectCancellable()` maybe helpful to sequence `then(() -> {})` and other events that run on the verify thread ... let me add them.

We will test both methods in our tests here.